pkg/controller/che/che_controller.go

```diff
@@ -376,49 +376,9 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{}, err
 	}
 
-	// create service accounts:
-	// che is the one which token is used to create workspace objects
-	// che-workspace is SA used by plugins like exec and terminal with limited privileges
-	cheServiceAccount := deploy.NewServiceAccount(instance, "che")
-	if err := r.CreateServiceAccount(instance, cheServiceAccount); err != nil {
-		return reconcile.Result{}, err
-	}
-	workspaceServiceAccount := deploy.NewServiceAccount(instance, "che-workspace")
-	if err := r.CreateServiceAccount(instance, workspaceServiceAccount); err != nil {
-		return reconcile.Result{}, err
-	}
-	// create exec and view roles for CheCluster server and workspaces
-	execRole := deploy.NewRole(instance, "exec", []string{"pods/exec"}, []string{"*"})
-	if err := r.CreateNewRole(instance, execRole); err != nil {
-		return reconcile.Result{}, err
-	}
-	viewRole := deploy.NewRole(instance, "view", []string{"pods"}, []string{"list"})
-	if err := r.CreateNewRole(instance, viewRole); err != nil {
-		return reconcile.Result{}, err
-	}
-	// create RoleBindings for created (and existing ClusterRole) roles and service accounts
-	cheRoleBinding := deploy.NewRoleBinding(instance, "che", cheServiceAccount.Name, "edit", "ClusterRole")
-	if err := r.CreateNewRoleBinding(instance, cheRoleBinding); err != nil {
+	if err := createServiceAccounts(instance, r); err != nil {
```

Comment:
Could we make the creation of the service account optional based on the enablement of openshift oauth? E.g. oauth disabled => create sa, oauth enabled => don't create the sa.

Reply:
are you sure that we don't need `che` sa at all when using oauth? it's used to run che pods https://github.com/eclipse/che-operator/blob/master/pkg/deploy/deployment_che.go#L73

Reply:
Hmm.. ok, that didn't occur to me :) I'm just worried about the "power" of that service account and would like to minimize the privs if the admin configures the server appropriately - namely, I don't want us to require namespace CRUD only when absolutely necessary.

Reply:
yes, I will try to just create `che` sa, but don't give it any privileges when oauth.